md/data/importer.py

```diff
@@ -1,56 +1,173 @@
 import logging
 import os
+import re
+
 import pandas as pd
 
 from django.conf import settings
-from django.db import connections
 
-from tsdata.util import call, line_count, download_and_unzip_data
+from tsdata.util import (call, download_and_unzip_data, get_csv_path,
+     get_datafile_path, line_count)
 
 
 logger = logging.getLogger(__name__)
-cursor = connections['traffic_stops_nc'].cursor()
+
+TIME_OF_STOP_re = re.compile(r'(\d?\d):(\d\d)( [AP]M)?$')
+DEFAULT_TIME_OF_STOP = '00:00'
+
+GENDER_MALE_re = re.compile(r'^(M|male|MALE|m +)$')
+GENDER_FEMALE_re = re.compile(r'^(F|female|w|F +)$')
+
+SEIZED_CONTRABAND_re = re.compile(r'^(Contraband.*|paraphernalia.*|Both)$')
+
+ETHNICITY_WHITE_re = re.compile(r'^(WHITE|W|W.)$')
+ETHNICITY_BLACK_re = re.compile(r'^(BLACK|BLK)$')
+ETHNICITY_TO_CODE = {
+    'HISPANIC': 'H',
+    'ASIAN': 'A',
+    'NATIVE AMERICAN': 'I',
+}
+
+STOP_REASON_re = re.compile(r'^ *(\d+) *- *(\d+) *\(.*\) *$')
+
+MD_COLUMNS_TO_DROP = (
+    'WHATSEARCHED', 'STOPOUTCOME', 'CRIME_CHARGED',
+    'REGISTRATION_STATE', 'RESIDENCE_STATE', 'MD_COUNTY',
+)
+
+
+def fix_ETHNICITY(s):
+    if ETHNICITY_WHITE_re.match(s):
+        return 'W'
+    elif ETHNICITY_BLACK_re.match(s):
+        return 'B'
+    else:
+        return ETHNICITY_TO_CODE.get(s, 'U')
+
+
+def fix_GENDER(s):
+    if GENDER_MALE_re.match(s):
+        return 'M'
+    elif GENDER_FEMALE_re.match(s):
+        return 'F'
+    else:
+        return 'U'
```

Comment:
In general, it might be nice to log these ``else`` statements across all the fix_* functions here just to note which values failed to normalize.